From c9619fbf2bd7640533c9dfa78e64ad1c57838918 Mon Sep 17 00:00:00 2001
From: Yumi Yukimura <me.cafebabe@gmail.com>
Date: Wed, 9 Oct 2024 17:15:07 +0800
Subject: [PATCH 7/7] flag_values: Enable AVF Custom VM support

* Makes it possible to run custom virtual machines. Refer to [1].
* Enable network too while at it (DICE and LLPVM are deps of network)

[1]: https://android.googlesource.com/platform/packages/modules/Virtualization/+/refs/tags/android-15.0.0_r3/docs/custom_vm.md

Change-Id: I5c677649edbf25b07076c3f857c9459df6becb87
---
 .../ap3a/RELEASE_AVF_ENABLE_DICE_CHANGES.textproto            | 4 ++++
 .../ap3a/RELEASE_AVF_ENABLE_LLPVM_CHANGES.textproto           | 4 ++++
 release/flag_values/ap3a/RELEASE_AVF_ENABLE_NETWORK.textproto | 4 ++++
 ...F_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES.textproto | 4 ++++
 4 files changed, 16 insertions(+)
 create mode 100644 release/flag_values/ap3a/RELEASE_AVF_ENABLE_DICE_CHANGES.textproto
 create mode 100644 release/flag_values/ap3a/RELEASE_AVF_ENABLE_LLPVM_CHANGES.textproto
 create mode 100644 release/flag_values/ap3a/RELEASE_AVF_ENABLE_NETWORK.textproto
 create mode 100644 release/flag_values/ap3a/RELEASE_AVF_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES.textproto

diff --git a/release/flag_values/ap3a/RELEASE_AVF_ENABLE_DICE_CHANGES.textproto b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_DICE_CHANGES.textproto
new file mode 100644
index 00000000..03d69f33
--- /dev/null
+++ b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_DICE_CHANGES.textproto
@@ -0,0 +1,4 @@
+name: "RELEASE_AVF_ENABLE_DICE_CHANGES"
+value: {
+  bool_value: true
+}
diff --git a/release/flag_values/ap3a/RELEASE_AVF_ENABLE_LLPVM_CHANGES.textproto b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_LLPVM_CHANGES.textproto
new file mode 100644
index 00000000..d4705ded
--- /dev/null
+++ b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_LLPVM_CHANGES.textproto
@@ -0,0 +1,4 @@
+name: "RELEASE_AVF_ENABLE_LLPVM_CHANGES"
+value: {
+  bool_value: true
+}
diff --git a/release/flag_values/ap3a/RELEASE_AVF_ENABLE_NETWORK.textproto b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_NETWORK.textproto
new file mode 100644
index 00000000..2594a9fe
--- /dev/null
+++ b/release/flag_values/ap3a/RELEASE_AVF_ENABLE_NETWORK.textproto
@@ -0,0 +1,4 @@
+name: "RELEASE_AVF_ENABLE_NETWORK"
+value: {
+  bool_value: true
+}
diff --git a/release/flag_values/ap3a/RELEASE_AVF_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES.textproto b/release/flag_values/ap3a/RELEASE_AVF_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES.textproto
new file mode 100644
index 00000000..7a25f56c
--- /dev/null
+++ b/release/flag_values/ap3a/RELEASE_AVF_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES.textproto
@@ -0,0 +1,4 @@
+name: "RELEASE_AVF_SUPPORT_CUSTOM_VM_WITH_PARAVIRTUALIZED_DEVICES"
+value: {
+  bool_value: true
+}
-- 
2.39.5

